extensions/replication/replicationStatusProcessor/ReplicationStatusProcessor.js

```diff
@@ -126,38 +125,32 @@ class ReplicationStatusProcessor {
     }
 
     /**
-     * Set the Redis hash key for each failed backend.
+     * Push any failed entry to the "retry" topic.
      * @param {QueueEntry} queueEntry - The queue entry with the failed status.
      * @param {Object} kafkaEntry - The kafka entry with the failed status.
-     * @param {Function} cb          [description]
+     * @param {Function} cb - The callback function
      * @return {undefined}
      */
-    _setFailedKeys(queueEntry, kafkaEntry, cb) {
+    _pushFailedEntry(queueEntry, kafkaEntry, cb) {
         const { status, backends } = queueEntry.getReplicationInfo();
         if (status !== 'FAILED') {
             return process.nextTick(cb);
         }
-        const bucket = queueEntry.getBucket();
-        const key = queueEntry.getObjectKey();
-        const versionId = queueEntry.getEncodedVersionId();
-        const fields = [];
-        backends.forEach(backend => {
-            const { status, site } = backend;
-            if (status === 'FAILED' && site === queueEntry.getSite()) {
-                const field = `${bucket}:${key}:${versionId}:${site}`;
-                const value = JSON.parse(kafkaEntry.value);
-                fields.push(field, JSON.stringify(value));
-            }
-            return undefined;
-        });
-        const cmds = ['hmset', redisKeys.failedCRR, ...fields];
-        return redisClient.batch([cmds], (err, res) => {
-            if (err) {
-                return cb(err);
-            }
-            const [cmdErr] = res[0];
-            return cb(cmdErr);
-        });
+        const backend = backends.find(b =>
+            b.status === 'FAILED' && b.site === queueEntry.getSite());
+        if (backend) {
+            const bucket = queueEntry.getBucket();
+            const key = queueEntry.getObjectKey();
+            const versionId = queueEntry.getEncodedVersionId();
+            const { site } = backend;
+            const message = {
+                field: `${bucket}:${key}:${versionId}:${site}`,
+                value: Buffer.from(kafkaEntry.value).toString(),
```

Comment:
Since moving away from storing object metadata in Redis, we can eliminate this.

Reply:
Will update in subsequent PRs.

Reply:
Will update in a subsequent PR so that we can maintain functionality when merging this one.